### ISSUE TITLE
Training log problems

### ISSUE THREAD

**Issue by lzane** [AlexeyAB/darknet#313]
I have some problems with the training logs.

> Region Avg IOU: 0.475194, Class: 1.000000, Obj: 0.187045, No Obj: 0.010328, Avg Recall: 0.500000,  **count**: 12
> 41372: 0.388809, **0.400530 avg**, 0.000010 rate, 2.823755 seconds, 2647808 images
Loaded: 6.112908 seconds

1. What's the meaning of count in the first line?
2. Doesn't the average loss calculate according to the loss in the first line and the scale in the .cfg file?

**Comment by AlexeyAB** [comment]
1. Count is a number of ground truth objects on the images from the current training batch:
    * count is printed: https://github.com/AlexeyAB/darknet/blob/75c39f57507ed81f33571271b939de400a69adf0/src/region_layer.c#L320
    * count is incremented: https://github.com/AlexeyAB/darknet/blob/75c39f57507ed81f33571271b939de400a69adf0/src/region_layer.c#L311
    * Check if next ground truth is present: https://github.com/AlexeyAB/darknet/blob/75c39f57507ed81f33571271b939de400a69adf0/src/region_layer.c#L262
    * Yolo set limit 30 of ground truth objects: https://github.com/AlexeyAB/darknet/blob/75c39f57507ed81f33571271b939de400a69adf0/src/region_layer.c#L259

----

> Doesn't the average loss calculate according to the loss in the first line **and the scale in the .cfg file**?

2. Average loss calculate according to the loss in the first line - in a such way `avg_loss = avg_loss*0.9 + loss*0.1;`: https://github.com/AlexeyAB/darknet/blob/75c39f57507ed81f33571271b939de400a69adf0/src/detector.c#L150-L151

But what `scale` do you mean? Explanation on cfg file parameters: https://github.com/AlexeyAB/darknet/issues/279#issuecomment-347002399

**Comment by MKYucel** [comment]
Hi @AlexeyAB,

just a quick question; can we change the count limit to something bigger than 30? It is apparent that we can from the source code you've mentioned, but what I am wondering is whether this would have an adverse effect on the training, can YOLO handle this? Do you have any experience on that?

**Comment by AlexeyAB** [comment]
@MKYucel Hi,

I didn't try to increase it. 
But you can try to do this.
Try to increase this value in several places:
* https://github.com/AlexeyAB/darknet/blob/75c39f57507ed81f33571271b939de400a69adf0/src/region_layer.c#L30
* https://github.com/AlexeyAB/darknet/blob/75c39f57507ed81f33571271b939de400a69adf0/src/region_layer.c#L190
* https://github.com/AlexeyAB/darknet/blob/75c39f57507ed81f33571271b939de400a69adf0/src/region_layer.c#L222
* https://github.com/AlexeyAB/darknet/blob/75c39f57507ed81f33571271b939de400a69adf0/src/region_layer.c#L259

Maybe you need to change something else.

**Comment by MKYucel** [comment]
Thanks, will give it a try soon.

**Comment by MKYucel** [comment]
Hello again,

I have tried changing the parts you've mentioned to 130 first, but regardless of the batch size and the number of GPUs i used, i could not get rid of getting NaNs therefore my training failed. Same happened for 50 but after several iterations.

I managed to pull it off with 40 by reducing the batch size drastically. Although I got some nan values here and there, the training finished with a lossv value actually lower than the one I had with 30. However, precision/recall values are far lower compared to one I had with 30. Just wanted to leave this comment here in case anyone else tries to do same thing.
